src/NuGet.Core/NuGet.Protocol/Providers/RepositorySignatureResourceProvider.cs

```diff
@@ -95,5 +98,13 @@ public RepositorySignatureResourceProvider()
 
             return null;
         }
+
+        private static string GenerateCacheKey(ServiceIndexEntry serviceEntry)
+        {
+            var index = serviceEntry.Type.IndexOf('/');
+            var version = serviceEntry.Type.Substring(index + 1).Trim();
+
+            return $"repository_signatures_{version}";
```

Comment:
👏

Reply:
Do you have a tracking issue for the same potential problem with the other resources? 
I did some quick looking and for example registration could hit the same problem.

Reply:
https://github.com/NuGet/Home/issues/8003